packages/node_modules/@ciscospark/redux-module-features/src/index.js

```diff
@@ -0,0 +1,45 @@
+import {Map} from 'immutable';
+
+export const ADD_FEATURE = `features/ADD_FEATURE`;
+
+export const initialState = new Map({
+  items: new Map()
+});
+
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+  case ADD_FEATURE:
+    return state.mergeIn([`items`], action.payload.feature);
+  default:
+    return state;
+  }
+}
+
+export function addFeature(feature) {
+  return {
+    type: ADD_FEATURE,
+    payload: {
+      feature
+    }
+  };
+}
+
+/**
+ * Gets the value of the requested feature toggle.
+ * @param {string} keyType <developer|user|entitlement>
+ * @param {string} key
+ * @param {object} spark
+ * @returns {Promise}
+ */
+export function getFeature(keyType, key, spark) {
```

Comment:
Should we have a "getAllFeatures" method to grab all the features for a user? Or is the preferred way to fetch individual ones?